test/PowerNotifications/PowerTests.cpp

```diff
@@ -0,0 +1,205 @@
+
+using PowerEventHandle =
+winrt::Windows::Foundation::EventHandler<winrt::Windows::Foundation::IInspectable>;
+
+using namespace winrt::Microsoft::ProjectReunion;
+
+namespace ProjectReunionPowerTests
+{
+    constexpr auto TIMEOUT = 2000;
+    class PowerTests
+    {
+    public:
+        BEGIN_TEST_CLASS(PowerTests)
+            TEST_CLASS_PROPERTY(L"ActivationContext", L"PowerTests.dll.manifest")
+            TEST_CLASS_PROPERTY(L"ThreadingModel", L"MTA")
+            END_TEST_CLASS()
+
+            TEST_METHOD(GetBatteryStatus)
+        {
+            auto stat = PowerManager::BatteryStatus();
+            VERIFY_ARE_EQUAL(stat, BatteryStatus::NotPresent);
+        }
+
+        TEST_METHOD(GetPowerSupplyStatus)
+        {
+            auto stat = PowerManager::PowerSupplyStatus();
+            VERIFY_ARE_EQUAL(stat, PowerSupplyStatus::Adequate);
+        }
+
+        TEST_METHOD(GetRemainingChargePercent)
+        {
+            auto stat = PowerManager::RemainingChargePercent();
+            VERIFY_ARE_EQUAL(stat, 100);
+        }
+
+        TEST_METHOD(CompositeBatteryStatusCallback)
+        {
+            // This test can only be run on systems/VM with a battery
+            // The values will need to be changed to reflect real-time values
+            BEGIN_TEST_METHOD_PROPERTIES()
+                TEST_METHOD_PROPERTY(L"Ignore", L"true")
+            END_TEST_METHOD_PROPERTIES()
+
+            wil::unique_handle event(CreateEvent(nullptr, false, false, nullptr));
+            THROW_LAST_ERROR_IF_NULL(event.get());
+            auto batteryStat = BatteryStatus::Discharging;
+            auto powerStat = PowerSupplyStatus::Inadequate;
+            int32_t remainingCharge = 15;
+            auto token = PowerManager::BatteryStatusChanged([&](const auto&, winrt::Windows::Foundation::IInspectable obj)
+                {
+                    batteryStat = PowerManager::BatteryStatus();
+                    powerStat = PowerManager::PowerSupplyStatus();
+                    remainingCharge = PowerManager::RemainingChargePercent();
+                    SetEvent(event.get());
+                });
+
+            VERIFY_IS_TRUE(WaitForSingleObject(event.get(), TIMEOUT) == WAIT_OBJECT_0);
+            VERIFY_ARE_EQUAL(batteryStat, BatteryStatus::Idle);
+            VERIFY_ARE_EQUAL(powerStat, PowerSupplyStatus::Adequate);
+            VERIFY_ARE_EQUAL(remainingCharge, 100);
+            PowerManager::BatteryStatusChanged(token);
+        }
+
+        TEST_METHOD(GetRemainingDischargeTime)
+        {
+            auto stat = PowerManager::RemainingDischargeTime();
+            VERIFY_ARE_EQUAL(stat.count(), -10000000);
+        }
+
+        TEST_METHOD(RemainingDischargeTimeCallback)
+        {
+            static wil::unique_handle event(CreateEvent(nullptr, false, false, nullptr));
+            THROW_LAST_ERROR_IF_NULL(event.get());
+
+            auto stat = winrt::Windows::Foundation::TimeSpan(std::chrono::seconds(1));
```

Comment:
`...TimeSpan(1s);`

std::chrono defines a bunch of funky operators so you can declare simple literals, e.g. `operator ""ms(unsigned long long ms)` is why `10ms` works. You can see the full set of chrono literal constant suffixes @ https://en.cppreference.com/w/cpp/chrono/duration

Reply:
TIL. Changed.